scripts/data_files/error.fmt

```diff
@@ -164,6 +164,7 @@ void mbedtls_strerror( int ret, char *buf, size_t buflen )
 
 #if defined(MBEDTLS_TEST_HOOKS)
 void (*mbedtls_test_hook_error_add)( int, int, const char *, int );
+void (*mbedtls_test_hook_value)( int, const char *, int );
```

Comment:
Why is this added here? I'd suggest to rather declare it in common.h.

Reply:
It's here on the pattern of the `(*mbedtls_test_hook_error_add)()` above, which was my source of inspiration for this solution. Now, after moving it into the `common.h` the linking fails badly:

```
[ 24%] Linking C executable rsa_sign_pss
/usr/bin/ld: ../../library/libmbedcrypto.a(asn1parse.c.o):(.bss+0x0): multiple definition of `mbedtls_test_hook_assert_test'; ../../CMakeFiles/mbedtls_test.dir/tests/src/certs.c.o:(.bss+0x0): first defined here
/usr/bin/ld: ../../library/libmbedcrypto.a(asn1parse.c.o):(.bss+0x40): multiple definition of `__odr_asan.mbedtls_test_hook_assert_test'; ../../CMakeFiles/mbedtls_test.dir/tests/src/certs.c.o:(.bss+0xcc): first defined here
/usr/bin/ld: ../../library/libmbedcrypto.a(bignum.c.o):(.bss+0x0): multiple definition of `mbedtls_test_hook_assert_test'; ../../CMakeFiles/mbedtls_test.dir/tests/src/certs.c.o:(.bss+0x0): first defined here
/usr/bin/ld: ../../library/libmbedcrypto.a(bignum.c.o):(.bss+0x40): multiple definition of `__odr_asan.mbedtls_test_hook_assert_test'; ../../CMakeFiles/mbedtls_test.dir/tests/src/certs.c.o:(.bss+0xcc): first defined here
/usr/bin/ld
....
[3800 lines of error messages removed]
```
Seems that the `error.fmt` gets converted into `error.c`, with additional extensions and where the new callback is prototyped.